src/Message/ApplePayAuthorizeRequest.php

```diff
@@ -0,0 +1,383 @@
+<?php
+
+namespace Omnipay\Vindicia\Message;
+
+use Omnipay\Common\Message\ResponseInterface;
+
+class ApplePayAuthorizeRequest extends \Omnipay\Common\Message\AbstractRequest
+{
+    /**
+     * The class to use for the response.
+     *
+     * @var string
+     */
+    protected static $RESPONSE_CLASS = '\Omnipay\Vindicia\Message\ApplePayAuthorizeResponse';
+
+    /**
+     * Default display name for Apple Pay Session.
+     *
+     * @var string
+     */
+    const DEFAULT_DISPLAY_NAME = 'DISPLAY_NAME_HERE';
+
+    /**
+     * @param array<string, mixed> $parameters
+     * @return ApplePayAuthorizeRequest
+     */
+    public function initialize(array $parameters = array())
+    {
+        if (!array_key_exists('displayName', $parameters)) {
+            $parameters['displayName'] = self::DEFAULT_DISPLAY_NAME;
+        }
+        parent::initialize($parameters);
+
+        return $this;
+    }
+
+    /**
+     * @return array
+     */
+    public function getData()
+    {
+        $this->validate('validationURL');
+
+        $data = array();
+
+        // These are from Apple's certificate – they should never change.
+        $data["merchantIdentifier"] = "merchant.com";
+        $data["initiative"] = "web";
+        $data["initiativeContext"] = "abcd.com";
+
+        if (isset($this->data['validationURL'])) {
+            $data['validationURL'] = $this->getValidationUrl();
+        }
+
+        if (isset($this->data['epochTimestamp'])) {
+            $data['epochTimestamp'] = $this->getTimeStamp();
+        }
+
+        if (isset($this->data['expiresAt'])) {
+            $data['expiresAt'] = $this->getExpirationTimeStamp();
+        }
+
+        if (isset($this->data['merchantSessionIdentifier'])) {
+            $data['merchantSessionIdentifier'] = $this->getMerchantSessionID();
+        }
+
+        if (isset($this->data['nonce'])) {
+            $data['nonce'] = $this->getNonceToken();
+        }
+
+        if (isset($this->data['merchantSessionIdentifier'])) {
+            $data['merchantSessionIdentifier'] = $this->getMerchantID();
+        }
+
+        if (isset($this->data['domainName'])) {
+            $data['domainName'] = $this->getDomainName();
+        }
+
+        if (isset($this->data['displayName'])) {
+            $data['displayName'] = $this->getDisplayName();
+        }
+
+        if (isset($this->data['signature'])) {
+            $data['signature'] = $this->getSignatureID();
+        }
+
+        return $data;
+    }
+
+    /**
+     * @return string
+     */
+    public function getValidationUrl()
+    {
+        return $this->getParameter('validationURL');
+    }
+
+    /**
+     * Set the validation URL for the authorize request
+     *
+     * @param string $url
+     * @return static
```

Comment:
Not sure about what to do for this docblock. It's setting a parameter but psalms complaining about the return type being too general. I've tried using Omnipay\Message\AbstractRequest like it suggested but that didn't work either.

Reply:
looks like that's what the other requests do... is it failing for all requests now due to a psalm update or something? if it's only failing on this one, that's a clue!

Reply:
It's failing for all of them.

Reply:
OK guess psalm updated. The ideal thing would be for Omnipay\Common\Message\AbstractRequest::setParameter to have a return type of `static`, then we'd be good to go. But since it doesn't, we'll have to use a typehint to tell psalm what's going on. so basically you need to put a line above the return statement that says `/** @var static */`, which will tell psalm what the return type is actually supposed to be.